### PR TITLE
py: use newer API

### PR DESCRIPTION
**Pull request by abhizer**
* introduces `SQLContext.sql` to set the DDL
* removes functions for setting up connectors

Is this a user-visible change (yes/no): yes

### CODE REVIEW

**Comment by snkas** [comment]
Technically we could also switch back to the `Pipeline` terminology instead of `SQLContext`? As they are nearly the same in the new API without all the other objects. Though not entirely sure.

**Review by ryzhyk — APPROVED** [review]

File: python/docs/feldera.rst [hunk 1]
```diff
@@ -33,18 +25,10 @@ feldera.output\_handler module
...
-------------------------
-
-.. automodule:: feldera.resources
-   :members:
-   :undoc-members:
-   :show-inheritance:
-
-feldera.sql\_schema module
---------------------------
+feldera.runtime\_config module
```
[the remark this answers is not in the document]
Comment (abhizer) [reply python/docs/feldera.rst:28]: The API also returns `program_config`, which although only has the compilation profile, I thought the distinction could be useful.

File: python/feldera/rest/feldera_client.py [hunk 2]
```diff
@@ -40,179 +37,32 @@ def __init__(
...
 
         try:
-            self.programs()
+            self.pipelines()
         except Exception as e:
             logging.error(f"Failed to connect to Feldera API: {e}")
             raise e
 
-    def programs(self) -> list[Program]:
+    def get_pipeline(self, pipeline_name):
```
[the remark this answers is not in the document]
Comment (abhizer) [reply python/feldera/rest/feldera_client.py:45]: Yes.

File: python/docs/feldera.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (snkas) [reply python/docs/feldera.rst:28]: There exists `program_config` as @abhizer mentioned, I think naming it `runtime_config` makes it unambiguous -- and also in line with the API, such that concepts between API and Python API do not need to be translated.

**Pull request merged**